R/calc_birdNBuse.R

```diff
@@ -16,25 +17,29 @@
 #' The function assumes that NestBoxID = LocationID.
 #'
 #' @param Brood_data Data frame. Data frame with standardized brood data.
-#' @return A data frame with the NestBoxID and corresponding FirstSeason and LastSeason.
+#' @return A data frame with all combinations of `PopID` and `NestBoxID`
+#'  and corresponding `StartSeason` and `EndSeason`.
 #' @export
 #'
 #' @examples
 #' library(dplyr)
-#' Brood_data <- tibble::tibble(BroodID = c(1:10), LocationID = LETTERS[c(1,3,1,1,2,3,1,2,1,2)],
-#'                              BreedingSeason = sort(sample(2012:2016, 10, replace = TRUE)))
-#' calc_birdNBuse(Brood_data)
+#' Brood_data <- tibble::tibble(BroodID = rep(c(1:5), 2),
+#'                              PopID = rep(c('XXX', 'YYY'), each = 5),
+#'                              LocationID = LETTERS[c(1,2,1,1,2,2,1,2,1,2)],
+#'                              BreedingSeason = c(2001, 2001, 2002, 2003, 2003,
+#'                              1962, 1963, 1963, 1965, 1965))
+#'calc_birdNBuse(Brood_data)
+
+
 calc_birdNBuse <- function(Brood_data){
 
   output <- Brood_data %>%
-    dplyr::group_by(LocationID) %>%
-    dplyr::summarise(StartSeason  = as.integer(min(BreedingSeason)),
-                  EndSeason  = as.integer(max(BreedingSeason))) %>%
+    dplyr::group_by(.data$PopID, .data$LocationID) %>%
+    dplyr::summarise(StartSeason  = as.integer(min(.data$BreedingSeason)),
+                  EndSeason  = as.integer(max(.data$BreedingSeason)),
+                  .groups = 'keep') %>%
     dplyr::ungroup()
```

Comment:
Great @LiamDBailey, thanks for this. You may have noticed my comment about something not working quite right yet with the `make_IndIdentifier()` function. If I recall correctly, issues arose when there were complimentary matches. For example, a bird with 2 captures, ringed at capture 1 with ring number "X1" and re-ringed at capture 2 to new ring number "X2", but both ring numbers reported like: 
CaptureNo           RING            RING2
1                           X1                X2
2                          X2                X1

I had already changed the function to prevent an "infinite loop" earlier, but could not figure out how to prevent it from including both complimentary pairs (X1-X2 and X2-X1) in the output.